src/BaseSlackHandler.ts

```diff
@@ -104,41 +106,47 @@ export interface ISlackFile {
     permalink?: string;
 }
 
+export interface ISlackUser {
+    id: string;
+    profile?: {
+        display_name?: string;
+        real_name?: string;
+        image_original?: string;
+        image_1024?: string;
+        image_512?: string;
+        image_192?: string;
+        image_72?: string;
+        image_48?: string;
+        bot_id?: string;
+    };
+}
+
 export abstract class BaseSlackHandler {
     constructor(protected main: Main) { }
 
-    public async getSlackRoomNameFromID(id: string, token: string) {
-        const channelsInfoApiParams = {
-            json: true,
-            qs: {
-                channel: id,
-                token,
-            },
-            uri: "https://slack.com/api/channels.info",
-        };
-        this.main.incRemoteCallCounter("channels.info");
+    public async getSlackRoomNameFromID(channel: string, client: WebClient) {
         try {
-            const response = await rp(channelsInfoApiParams);
+            const response = (await client.conversations.info({ channel })) as ConversationsInfoResponse;
             if (response && response.channel && response.channel.name) {
-                log.info(`channels.info: ${id} mapped to ${response.channel.name}`);
+                log.info(`conversations.info: ${channel} mapped to ${response.channel.name}`);
                 return response.channel.name;
             }
-            log.info("channels.info returned no result for " + id);
+            log.info("conversations.info returned no result for " + channel);
         } catch (err) {
-            log.error("Caught error handling channels.info:" + err);
+            log.error("Caught error handling conversations.info:" + err);
         }
-        return id;
+        return channel;
     }
 
-    public async doChannelUserReplacements(msg: ISlackMessage, text: string|undefined, token: string) {
+    public async doChannelUserReplacements(msg: ISlackMessage, text: string|undefined, webClient: WebClient) {
         if (text === undefined) {
             return;
         }
-        text = await this.replaceChannelIdsWithNames(msg, text, token);
-        return await this.replaceUserIdsWithNames(msg, text, token);
+        text = await this.replaceChannelIdsWithNames(msg, text, webClient);
+        return await this.replaceUserIdsWithNames(msg, text);
     }
 
-    public async replaceChannelIdsWithNames(message: ISlackMessage, text: string, token: string) {
+    public async replaceChannelIdsWithNames(message: ISlackMessage, text: string, slackClient: WebClient) {
```

Comment:
why the mixing of `webClient` and `slackClient` shouldn't we pick one?

Reply:
oh yeah, I think I was being indecisive. probably should use slackClient